### ISSUE TITLE
Build Failed:  unknown value ‘native’ for -march

### ISSUE THREAD

**Issue by MakingL** [chenshuo/muduo#446]
master 分支 CentOS7.6 arm-64 运行 `./build.sh` 构建失败，找不到原因

![image1](https://user-images.githubusercontent.com/31580426/77305003-41526c80-6d30-11ea-925e-59de7563dbed.png)

注释掉编译选项中的 `-march=native`，还是编译通不过

![image2](https://user-images.githubusercontent.com/31580426/77305171-8aa2bc00-6d30-11ea-8812-441fe450e8d0.png)

## Linux distro and version? x86 or ARM? 32-bit or 64-bit?

- CentOS7.6 
- ARM 64-bit

![image3](https://user-images.githubusercontent.com/31580426/77305415-f2590700-6d30-11ea-8058-97807b86ce69.png)

## Branch (cpp98/cpp11/cpp17) and version of muduo?

- master 分支

## Version of cmake, gcc and boost? (If not from distro.)

- gcc 4.8.5
- make 3.82
- cmake 2.8.12.2

![image](https://user-images.githubusercontent.com/31580426/77305579-3b10c000-6d31-11ea-8498-99cc633f95a7.png)

**Comment by chenshuo** [comment]
Try changing kMaxNumericSize to be `std::numeric_limits<long double>::digits10 + 11`.
Looks like your ARM 64 has 128-bit `long double` type.

**Comment by MakingL** [comment]
> Try changing kMaxNumericSize to be `std::numeric_limits<long double>::digits10 + 11`.
> Looks like your ARM 64 has 128-bit `long double` type.

I checked the value of `std :: numeric_limits <long double> :: digits10` on my host, it is 33. And I set ` kMaxNumericSize` in `LogStream.h` line 168  to 44, it is built passed and everything works fine. thank you.
In addition, it seems GCC 8 where  `-march=native` is unsupported On ARM/AArch64, so I removed the `-march=native` option in CMakeLists.txt.
